test/test_optim.py

```diff
@@ -685,6 +673,25 @@ def test_step_lr(self):
         scheduler = StepLR(self.opt, gamma=0.1, step_size=3)
         self._test(scheduler, targets, epochs)
 
+    def test_get_lr_step_lr(self):
+        from torch.nn import Parameter
+        epochs = 10
+        optimizer = torch.optim.SGD([Parameter(torch.randn(2, 2, requires_grad=True))], 0.1)
+        targets = [[0.1] * 3 + [0.01] * 3 + [0.001] * 3 + [0.0001]]
+        scheduler = torch.optim.lr_scheduler.StepLR(optimizer, 3, gamma=0.1)
+        self._test_get_lr(scheduler, targets, epochs)
+
+    def test_get_lr_multi_step_lr(self):
+        # lr = 0.05     if epoch < 2
+        # lr = 0.005    if 2 <= epoch < 5
+        # lr = 0.0005   if epoch < 9
+        # lr = 0.00005   if epoch >= 9
+        epochs = 10
+        single_targets = [0.05] * 2 + [0.005] * 3 + [0.0005] * 4 + [0.00005] * 3
```

Comment:
why [0.00005] * 3 instead of [0.00005] * 2 ?

Reply:
The test checks at each one of the 10 iterations, so the test still pass with a longer target list. I'll still make it `[0.00005] * 1` for clarity.